grafana/loki
# Implement POC for logs-filtering

This PR is intended to demonstrate an approach to query-time filtering of log streams. In the approach, deletes are attached to an incoming request that is pushed down to the storage and ingester layers. The layers independently assemble their pipelines and filters -- just like now -- but extra filtering is added for streams and lines that match the delete expression.

This doesn't include:
1.  dynamic or multiple deletes -- there's only one hard coded, for now
2. sample queries

#### To Test:
1. Add logs with `tools/populate-logs.sh` (there are only 4)
1. query for `{label1="foo"}`
1. uncomment various cases [here](https://github.com/grafana/loki/blob/4eb99b668f4f32bd5321ef8ff275f003712a8fbf/pkg/querier/querier.go#L110) to see lines filtered from the query.
